mozci/task.py

```diff
@@ -164,10 +165,35 @@ def update_group(result):
                 if not is_bad_group(self.id, result.group)
             ]
 
+        # I'm storing in the cache in here rather than at the end of _load_errorsummary()
+        # because there's various modifications of various self properties.
+        # cachy's put() overwrites the value in the cache; add() would only add if its empty
+        # Only store data if there's something to store
+        if self._errors or self._groups or self._results:
+            logger.debug("Storing {} in the cache".format(self.id))
+            adr.config.cache.put(
+                self.id,
+                {
+                    "errors": self._errors,
+                    "groups": self._groups,
+                    "results": self._results,
+                },
+                1,  # XXX: What should the value in minutes be?
+            )
+
     def _load_errorsummary(self):
+        # I don't believe we need to invalidate the data since once we parse
+        # the error summary artifact we should not expect a different result in
+        # the future since the artifact doesn't change
```

Comment:
The only problem is if we change the format of the properties of the tasks

Reply:
Or e.g. if we add more properties

Reply:
Do you have any suggestions as to how or when to invalidate the data?
A function that clears the cache for all the tasks associated to a push?

Reply:
I have no idea :(